README.md

```diff
@@ -1,2 +1,108 @@
+CODE_ANALYSIS_LOGIN    | The analysis server account to use when analysing or publishing 
+CODE_ANALYSIS_PASSWORD | The analysis server account password                     
+
+---
+
+## Usage
+
+### Configuration
+
+The following configuration variables are required to be set, whether via Environment Variables or your project's `application.yml` file:
+
+Environment Variable Key    | application.yml Key       | Example                                                     
+--------------------------- | ------------------------- | -------------
+KAFKA_BROKER_ADDR           | kafka.broker.addr         | kafka:9092
+KAFKA_CONFIG_ACKS           | kafka.config.acks         | WAIT_FOR_ALL
+KAFKA_CONFIG_RETRIES        | kafka.config.retries      | false
+KAFKA_CONFIG_IS_ROUND_ROBIN | kafka.config.isRoundRobin | 10
+
+### Building An Email
+
+The `EmailData` object defines the minimum required data to send an email - `to` (the intended recipient's email address), `subject` and `cdnHost`. To add application data to an email, a new class should be defined which extends `EmailData` and includes attributes for storing the data. For example, from [`Suppression Api`](https://github.com/companieshouse/suppression-api):
+
+```java
+import uk.gov.companieshouse.kafka_email.EmailData;
+
+public class ApplicationReceivedEmailData extends EmailData {
+
+    private Suppression suppression;
+    ...
+
+}
+```
+
+Once this data object has been defined, it can be used to construct an `EmailSend` object like so:
+```java
+import uk.gov.companieshouse.kafka_email.EmailFactory;
+import uk.gov.companieshouse.kafka_email.EmailSend;
+
+...
+
+private final EmailFactory emailFactory;
+
+...
+
+final EmailSend email = emailFactory.buildEmail(
+                emailData, EMAIL_APP_ID, messageType);
+```
+where:
+* `emailData` (_EmailData_) is an instance of the data object defined in the previous step
+* `EMAIL_APP_ID` (_String_) is the service application's name (e.g. `suppression-api`); and
+* `messageType` (_String_) references an email template in the CH Notification API's template registry.
+
+> **Note:** The call to this method will need to handle `JsonProcessingException` and `SerializationException`.
+
+### Sending An Email
+
+Providing the previous steps have been followed, sending the email to Kafka is very simple:
+```java
+import uk.gov.companieshouse.kafka_email.EmailKafkaProducer;
+
+...
+
+private final EmailKafkaProducer emailKafkaProducer;
+
+...
+
+emailKafkaProducer.sendEmail(email);
+```
+
+> **Note:** The call to this method will need to handle `ExecutionException` and `InterruptedException`.
+
+## Spring Applications
+
+`EmailFactory` and `EmailKafkaProducer` have been specified as Spring Services. In Spring applications, these classes do not require explicit instantiation. See [`Suppression Api`](https://github.com/companieshouse/suppression-api)'s `EmailService` for a full usage example.
```

Comment:
What is `'s` for? It renders wired.

Reply:
Just to specify that the EmailService belongs to the Suppression API. I'll rephrase.